### ISSUE TITLE
Issues when recategorizing custom nodes

### ISSUE THREAD

**Issue by andydandy74** [DynamoDS/Dynamo#5292]
After editing (and saving) a custom node multiple times I noticed that several instances of it appeared in the node browser. They disappear after restarting Dynamo.
![multiple instances](https://cloud.githubusercontent.com/assets/3014437/9842523/5b7f0138-5aaf-11e5-83af-07cdb8473be9.PNG)
0.8.3.2553 Revit addin (haven't tried to reproduce in sandbox)

**Comment by andydandy74** [comment]
Something similar happens when recategorizing a node. The node appears twice in the new category but also appears once at its own location:
![recategorization](https://cloud.githubusercontent.com/assets/3014437/9896682/fa779138-5c3f-11e5-9304-36ffcb10e4d5.PNG)
FYI @Racel

**Comment by Racel** [comment]
Thanks, @andydandy74. Looks like a bug. We will look into it.

FYI @jnealb

**Comment by jnealb** [comment]
@andydandy74 Can not repro the cloned dyf syndrome with 0.8.3.2585 running on Revit. May need exact steps.

**Comment by andydandy74** [comment]
@jnealb - You're right, neither can I. Maybe a glitch. What I still see is weird behaviour of nodes in the browser when recategorizing: 
- Recategorizing nodes that live in the definitions folder: The node apperas in the old category **and** the new category
- Recategorizing nodes that live in a package: The node does not move to the new category (only after restarting Dynamo) - I think this one's been around for a while actually.

0.8.3.2580

**Comment by jnealb** [comment]
@andydandy74 We can reproduce the re-categorization issue. Will advise.

**Comment by jnealb** [comment]
Tracked with MAGN-8421.

**Comment by smangarole** [comment]
@andydandy74 Can you confirm if this is still happening in Dynamo 1.3? Please re-open if reproducible.

**Comment by andydandy74** [comment]
@smangarole 
* **Solved**: Saving a custom node multiple times will **not** make it appear multiple times in the library any more. 
* **Still an issue**: Recategorizing nodes that live in a package will only make them appear in their new category in the  library **after** restarting Dynamo.
* **Solved**: Recategorizing nodes that live outside a package now works the same way as recategorizing nodes in packages.
* **Still an issue**: When saving a custom node under a new name, the new node will only appear in the library **after** restarting Dynamo.

So basically the **only unsolved issue** here is that the **library view doesn't refresh** for newly created or duplicated ("saved as") nodes.

Also, I cannot reopen this issue (repo maintainers only, it seems).
